### PR TITLE
feat: implement trim_whitespace of mustache

### LINKED ISSUE #34: Implement mustache.rs
Is there anybody working on it? If not I will do it.

### PR DESCRIPTION
**Pull request by linrium**
Fixed #34

### CODE REVIEW

**Comment by linrium** [comment]
CI failed because of the conflict between `Rc<RefCell>` and getting the reference. Fix it later

File: src/compiler/interfaces.rs [hunk 1]
```diff
@@ -1,19 +1,31 @@
...
 use strum_macros::Display;
+use swc_ecma_ast::Str;
 use swc_estree_ast::{AssignmentExpression, Program};
 
 use super::node::Node;
 
-#[derive(Clone)]
+pub type Children = Rc<RefCell<Vec<Rc<RefCell<TemplateNode>>>>>;
+
+pub trait GetChildren {
```
Comment (qwertydelle) [inline src/compiler/interfaces.rs:16]: I suggest deleting this trait and adding its functions to the `TmpNode` trait.

**Comment by pintariching** [comment]
I pushed some fixes for the errors. I didn't think I could push directly to your fork, I should've probably your branch into the main repo or opened a PR on your fork, sorry about that. You can revert the commits if you were working on a different solution!

**Comment by linrium** [comment]
No problem. I think we need to discuss more how we mutable inner value. Previously I added lifetime and returned the &’a mut A. Still, we can't use this variable anymore after mutable borrowing, so I tried Rc and RefCell, but it's pretty lengthy.

**Comment by linrium** [comment]
By the way, I think prop_name should be Hashmap so we can get it in O(1)

**Comment by linrium** [comment]
No problem. Thanks for your help.

**Comment by pintariching** [comment]
This seems ok for now, do I merge it or is there anything else to do?

**Comment by linrium** [comment]
I've just added some unit tests

**Pull request merged**